examples/api/lib/material/date_picker/custom_calendar_date_picker.0.dart

```diff
@@ -0,0 +1,148 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/material.dart';
+
+/// Flutter code sample demonstrating how to use a custom [CalendarDelegate]
+/// with [CalendarDatePicker] to implement a hypothetical calendar system
+/// where even-numbered months have 21 days, odd-numbered months have 28 days,
+/// and every month starts on a Monday.
+
+void main() => runApp(const CalendarDatePickerApp());
+
+class CalendarDatePickerApp extends StatelessWidget {
+  const CalendarDatePickerApp({super.key});
+
+  @override
+  Widget build(BuildContext context) {
+    return const MaterialApp(home: CalendarDatePickerExample());
+  }
+}
+
+class CalendarDatePickerExample extends StatefulWidget {
+  const CalendarDatePickerExample({super.key});
+
+  @override
+  State<CalendarDatePickerExample> createState() => _CalendarDatePickerExampleState();
+}
+
+class _CalendarDatePickerExampleState extends State<CalendarDatePickerExample> {
+  DateTime? selectedDate;
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      appBar: AppBar(title: const Text('Custom Calendar')),
+      body: Column(
+        spacing: 16,
+        children: <Widget>[
+          CalendarDatePicker(
+            initialDate: DateTime(2025, 2, 8),
+            firstDate: DateTime(2025),
+            lastDate: DateTime(2026),
+            onDateChanged: (DateTime pickedDate) {
+              setState(() {
+                selectedDate = pickedDate;
+              });
+            },
+            calendarDelegate: const CustomCalendarDelegate(),
+          ),
+          const Divider(height: 1),
+          Text(
+            selectedDate != null
+                ? '${selectedDate!.day}/${selectedDate!.month}/${selectedDate!.year}'
+                : 'No date selected',
+          ),
+        ],
+      ),
+    );
+  }
+}
+
+/// A custom calendar system where even-numbered months have 21 days,
+/// odd-numbered months have 28 days, and every month starts on a Monday.
+///
+/// This hypothetical calendar follows a fixed structure:
+/// - **Even-numbered months (2, 4, 6, etc.)** always have **21 days**.
+/// - **Odd-numbered months (1, 3, 5, etc.)** always have **28 days**.
+/// - **The first day of every month is always a Monday**, ensuring a consistent weekly alignment.
+class CustomCalendarDelegate extends CalendarDelegate {
```

Comment:
nit: should this be `extends CalendarDelegate<DateTime>`?

Reply:
Done